test/matlab_experimental/ctTestThermo.m

```diff
@@ -0,0 +1,482 @@
+
+            self.assumeMatches(self.phase.eosType, 'ideal-gas');
+            self.assumeTrue(self.phase.isIdealGas);
+
+            val = self.phase.X;
+            exp = zeros(1, 10);
+            exp(1) = 1.0;
+            tol = ones(1, 10).*self.rtol;
+            self.assumeEqual(val, exp, 'RelTol', tol);
+
+            val = self.phase.Y;
+            self.assumeEqual(val, exp, 'RelTol', tol);
+
+            val1 = [self.phase.H, self.phase.S, ...
+                    self.phase.U, self.phase.G, ...
+                    self.phase.cp, self.phase.cv];
+            self.phase.basis = 'mass';
+            val2 = [self.phase.H, self.phase.S, ...
+                    self.phase.U, self.phase.G, ...
+                    self.phase.cp, self.phase.cv];
+            exp = val2.*self.phase.meanMolecularWeight;
+            tol = ones(1, 9).*self.rtol;
+            self.assumeEqual(val1, exp, 'RelTol', tol);
+
+            val = self.phase.isothermalCompressibility;
+            exp = 1.0 / self.phase.P;
+            self.assumeEqual(val, exp, 'RelTol', self.rtol);
+
+            val = self.phase.thermalExpansionCoeff;
+            exp = 1.0 / self.phase.T;
+            self.assumeEqual(val, exp, 'RelTol', self.rtol);
+
+        end
+
+        function testGetStateMole(self)
+            self.phase.TDX = {350.0, 0.01, 'H2:0.1, O2:0.3, AR:0.6'};
+            self.checkGetters;
+        end
+
+        function testGetStateMass(self)
+            self.phase.basis = 'mass';
+            self.phase.TDY = {350.0, 0.7, 'H2:0.1, H2O2:0.1, AR:0.8'};
+            self.checkGetters;
+        end
+
+        function testSetComposition(self)
+            xx = zeros(1, self.phase.nSpecies);
+            xx(3) = 1.0;
+            self.phase.X = xx;
+            yy = self.phase.Y;
+            tol = ones(1, 10).*self.atol;
+
+            self.assumeEqual(xx, yy, 'AbsTol', tol)
+        end
+
+        function testSetCompositionBadLength(self)
+            xx = zeros(1, 5);
+            self.setInvalidValue('X', [], 'cannot be empty');
+            self.setInvalidValue('X', xx, 'must be equal');
+            self.setInvalidValue('Y', xx, 'must be equal');
+        end
+
+        function testSetCompositionString(self)
+            self.phase.X = 'h2:1.0, o2:1.0';
+            xx = self.phase.X;
+
+            self.assumeEqual(xx(1), 0.5, 'AbsTol', self.atol);
+            self.assumeEqual(xx(4), 0.5, 'AbsTol', self.atol);
+        end
+
+        function testSetCompositionStringBad(self)
+            self.setInvalidValue('X', 'H2:1.0,CO2:1.5', 'Unknown species');
+            self.setInvalidValue('X', 'H2:1.0,O2:asdf', 'Trouble processing');
+            self.setInvalidValue('X', 'H2:1.e-x4', 'Trouble processing');
+            self.setInvalidValue('X', 'H2:1e-1.4', 'decimal point in exponent');
+            self.setInvalidValue('X', 'H2:0.5,O2:1.0,H2:0.1', 'Duplicate key');
+            self.setInvalidValue('X', '', 'cannot be empty');
+        end
```

Comment:
This is an example of what I meant that for the Matlab toolbox, you do not need to test all the behavior of the underlying library. Here, I think all you need to to is test the different places an error can come from once. That is, the "cannot be empty" error, which is raised from the Matlab setter, and just one of the others, all of which are raised by the C++ `setMassFractionsByName` or `parseCompString` methods. The rest of these cases are adequately handled by the Python and C++ test suites.